python/lsst/ip/isr/isrMock.py

```diff
@@ -0,0 +1,915 @@
+
+        Parameters
+        ----------
+        ampData : `lsst.afw.image.ImageF`
+            Amplifier image to operate on.
+        scale : float
+            Peak flux of the source to add.
+
+        Notes
+        -----
+        This simulates simple crosstalk between amplifiers.
+        """
+        for x in range(0, ampDataSource.getDimensions().getX()):
+            for y in range(0, ampDataSource.getDimensions().getY()):
+                ampDataTarget.getArray()[y][x] = (ampDataTarget.getArray()[y][x] +
+                                                  scale * ampDataSource.getArray()[y][x])
+
+    # Functional form data values
+    def amplifierAddFringe(self, amp, ampData, scale):
+        r"""Add a fringe-like ripple pattern to an amplifier's image data.
+
+        Parameters
+        ----------
+        amp : `lsst.afw.ampInfo.AmpInfoRecord`
+            Amplifier to operate on.  Needed for amp<->exp coordinate transforms.
+        ampData : `lsst.afw.image.ImageF`
+            Amplifier image to operate on.
+        scale : float
+            Peak intensity scaling for the ripple.
+
+        Notes
+        -----
+        This uses an offset sinc function to generate a ripple
+        pattern.  True fringes have much finer structure, but this
+        pattern should be visually identifiable.  The (x, y)
+        coordinates are in the frame of the amplifier, and (u, v) in
+        the frame of the full trimmed image.
+        """
+        for x in range(0, ampData.getDimensions().getX()):
+            for y in range(0, ampData.getDimensions().getY()):
+                (u, v) = self.localCoordToExpCoord(amp, x, y)
+                ampData.getArray()[y][x] = (ampData.getArray()[y][x] +
+                                            scale * np.sinc(((u - 100)/50)**2 + (v / 50)**2))
+
+    def amplifierAddFlat(self, amp, ampData, size):
+        r"""Add a flat-like dome pattern to an amplifier's image data.
+
+        Parameters
+        ----------
+        amp : `lsst.afw.ampInfo.AmpInfoRecord`
+            Amplifier to operate on.  Needed for amp<->exp coordinate transforms.
+        ampData : `lsst.afw.image.ImageF`
+            Amplifier image to operate on.
+        size : float
+            Size of the dome to use.  A larger value results in a
+            smaller center-to-edge fall off.
+
+        Notes
+        -----
+        This uses a centered 2-d Gaussian to simulate an illumination
+        pattern that falls off towards the edge of the detector.  The
+        e(x, y) coordinates are in the frame of the amplifier, and (u, v)
+        in the frame of the full trimmed image.
+        """
+        for x in range(0, ampData.getDimensions().getX()):
+            for y in range(0, ampData.getDimensions().getY()):
+                (u, v) = self.localCoordToExpCoord(amp, x, y)
+                f = np.exp(-0.5 * ((u - 100)**2 + (v - 100)**2) / size**2)
+                ampData.getArray()[y][x] = (ampData.getArray()[y][x] * f)
```

Comment:
The function name, and the docstring say "add", but this is a multiplicative effect. I see how this is adding an effect, so I don't disagree, but given that a lot of things in here are actually being added (all the noises, the dark current etc), I think it would be good to be clear.

Also, I think the function should probably be renamed to have `dome` in it somewhere, lest it get confused with normal flat stuff.

Reply:
I've changed to `amplifierMultiplyFlat` to be mathematically consistent, and removed all mentions of "dome" for clarity.